src/ml-search.service.js

```diff
@@ -767,13 +769,15 @@
           return $q.reject(new Error('No constraint exists matching ' + facetName));
         }
 
-        var newOptions = { constraint: constraint, values: constraint };
-        newOptions.values['values-option'] = newOptions.values.range['facet-option'];
+        constraint = _.cloneDeep(constraint);
+
+        var newOptions = { constraint: _.cloneDeep(constraint), values: constraint };
```

Comment:
Are both `_.cloneDeep()` invocations necessary? The original constraint object isn't used again, so I would think one would be enough ...